tools/commands-packages.js

```diff
@@ -276,6 +289,63 @@ main.registerCommand({
     return 1;
   }
 
+  // If we just want to update the package metadata, then we have all we
+  // need. Don't bother building the package.
+  if (options.update) {
+    var name = packageSource.name;
+    var version = packageSource.version;
+
+    // You can't change the metadata of a record that doesn't exist.
+    var existingRecord =
+          catalog.official.getVersion(packageSource.name, packageSource.version);
+    if (! existingRecord) {
+      Console.error(
+        "You can't call",  Console.command("`meteor publish --update`"),
+        "on version " + version + " of " + "package '" + name +
+          "' without publishing it first.");
+      return 1;
+    }
+
+    // Load in the user's documentation, and check that it isn't blank.
+    var readmeInfo;
+    main.captureAndExit(
+      "=> Errors while publishing:", "reading documentation",
+      function () {
+       readmeInfo = packageSource.processReadme();
+    });
+
+    // You are still not allowed to upload a blank README.md.
+    if (readmeInfo && readmeInfo.hash === files.blankHash) {
+      Console.error(
+        "Please fill it out your documentation file! If you don't want to",
+        "update new documentation, you can remove it out from the package",
```

Comment:
All right. I think the new and more sensible rule, is that 'publish --update' works exactly like 'publish', minus the actually uploading builds/sources etc. So, the git/summary you have in package.js is the git/summary you get, and so on. 

I don't know, I guess that makes way more sense and I was totally overthinking various non-existent failure modes.